subsys/bluetooth/controller/ll_sw/ull_slave.c

```diff
@@ -77,6 +78,46 @@ void ull_slave_setup(memq_link_t *link, struct node_rx_hdr *rx,
 
 	/* Populate the slave context */
 	pdu_adv = (void *)((struct node_rx_pdu *)rx)->pdu;
+
+	peer_addr_type = pdu_adv->tx_addr;
+	memcpy(peer_addr, pdu_adv->connect_ind.init_addr, BDADDR_SIZE);
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	uint8_t rl_idx = ftr->rl_idx;
+
+	if (rl_idx != FILTER_IDX_NONE) {
+		/* Get identity address */
+		ll_rl_id_addr_get(rl_idx, &peer_addr_type, peer_id_addr);
+		/* Mark it as identity address from RPA (0x02, 0x03) */
+		peer_addr_type += 2;
+	} else {
+#else /* CONFIG_BT_CTLR_PRIVACY */
+	if (1) {
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+		memcpy(peer_id_addr, peer_addr, BDADDR_SIZE);
+	}
+
+#if defined(CONFIG_BT_CTLR_IGNORE_SAME_PEER_CONN)
+	uint8_t own_addr_type = pdu_adv->rx_addr;
+	uint8_t *own_addr = ll_addr_get(own_addr_type, NULL);
```

Comment:
Why not use the `pdu_adv->connect_ind.adv_addr`?

Reply:
The intention is to ignore connections to the same identity; `pdu_adv->connect_ind.adv_addr` may be an RPA that is refreshed periodically without changing identity.

Reply:
My concern is regarding extended advertising where each advertising set has its own random address.

Reply:
Good point. I will add a field to `ll_adv_set` to avoid races between different advertising sets.